### PR TITLE
Add notify_in_progress_reviewers to config and PR functionality

### PR DESCRIPTION
**Pull request by awortham**
This PR introduces a new config option called notify_in_progress_reviewers. The problem to solve is this: 
1. You submit a PR
2. Two different reviewers write comments about the code. They don't have to request changes. They simply put in a comment and then do not click the approve button. 
3. You make changes and then wait.
4. Reviewbot believes this PR is in progress and no longer sends out notifications for this PR.
5. You have to manually request reviewers to come back and take a look.

This change will allow each app/team to decide whether they like this functionality or not. If they want to continue to receive notifications until the PR gets reviewed then can a line to their config and the behavior will change. If they do not, the bot will default to the original behavior.

### CODE REVIEW

File: spec/pull_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+describe ReviewBot::PullRequest do
+  before do
+    JSON.parse(config).each do |app, app_config|
+      @owner, @repo = app.split('/')
+      @app_config = app_config
+    end
+
+    GH = Github.new(oauth_token: ENV['GH_AUTH_TOKEN'])
+
+    allow_any_instance_of(Github::Client::PullRequests).to receive(:list).and_wrap_original do |m, *args|
```
Comment (danielma) [inline spec/pull_request_spec.rb:12]: Personal preference: I prefer webmock to `allow_any_instance_of`

File: spec/pull_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      context 'ignore in progress is false' do
+        let(:config) { {'ministrycentered/reviewbot': { 'ignore_in_progress': false }}.to_json }
+
+        it 'is true' do
+          expect(@pull.needs_review?).to eq false
+        end
+      end
+
+      context 'ignore in progress is true' do
+        let(:config) { {'ministrycentered/reviewbot': { 'ignore_in_progress': true }}.to_json }
```
Comment (danielma) [inline spec/pull_request_spec.rb]: If I'm understanding this config properly, it means that we should notify the channel even if there are reviews in progress. I don't feel like `ignore_in_progress` maps very well to that intent. Are there any other names you considered? I might try something like `notify_in_progress_reviewers`

Comment (awortham) [reply spec/pull_request_spec.rb]: 👌 sounds good. I'll make that change.

**Review by danielma — APPROVED** [review]

**Comment by awortham** [comment]
@danielma Do you need anything else from me on this? (I don't have write access to this repo so I cannot merge) If not, I'll leave it in your capable hands.

**Pull request closed without merging**